source/Halibut.Tests/ExceptionContractFixture.cs

```diff
@@ -0,0 +1,290 @@
+                .AsLatestClientAndLatestServiceBuilder()
+                .WithStandardServices()
+                .WithHalibutTimeoutsAndLimits(halibutTimeoutsAndLimits)
+                .Build(CancellationToken);
+            
+            var client = clientAndService.CreateAsyncClient<IEchoService, IAsyncClientEchoServiceWithOptions>(point =>
+            {
+                // We need to use a non localhost address to get the correct exception
+                return new ServiceEndPoint(new Uri("https://20.5.79.31:10933/"), point.RemoteThumbprint, point.Proxy, clientAndService.Client!.TimeoutsAndLimits);
+            });
+
+            var cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(5));
+
+            (await AssertException.Throws<HalibutClientException>(async () => await client.SayHelloAsync("Hello", new(cancellationTokenSource.Token, CancellationToken.None))))
+                .And.Message.Should().Contain("An error occurred when sending a request to 'https://20.5.79.31:10933/', after the request began: The operation was canceled.");
+        }
+
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testNetworkConditions: false, testPolling: false, testWebSocket: false)]
+        public async Task WhenTheListeningRequestIsCancelledWhileConnecting_AndTheConnectionIsEstablishedButPaused_AHalibutClientExceptionShouldBeThrown(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            var halibutTimeoutsAndLimits = new HalibutTimeoutsAndLimitsForTestsBuilder().Build();
+            halibutTimeoutsAndLimits.TcpClientConnectTimeout = TimeSpan.FromSeconds(60);
+            halibutTimeoutsAndLimits.ConnectionErrorRetryTimeout = TimeSpan.FromSeconds(600);
+            halibutTimeoutsAndLimits.RetryCountLimit = int.MaxValue;
+
+            await using var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                .AsLatestClientAndLatestServiceBuilder()
+                .WithStandardServices()
+                .WithHalibutTimeoutsAndLimits(halibutTimeoutsAndLimits)
+                .WithPortForwarding(out var portForwarder)
+                .Build(CancellationToken);
+            
+            portForwarder.Value.EnterPauseNewAndExistingConnectionsMode();
+            var client = clientAndService.CreateAsyncClient<IEchoService, IAsyncClientEchoServiceWithOptions>();
+
+            var cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(5));
+
+#if NETFRAMEWORK
+            // net48 does not support cancellation of the request as the DeflateStream ends up using Begin and End methods which don't get passed the cancellation token
+            // This results in a different exception message / a different code path executing. This highlights an inconsistency in the response from 
+            // Halibut due to the way the code is written, so leaving this here as it should really be consistent.
```

Comment:
Is this something we need to address?

Reply:
It's a known issue and currently, only TentacleClient will cancel requests and that is always and can only be net6.0+ so this is "OK". Support for net48 should hopefully be on the way out as well so we probably won't ever need to fix this. If we do we can.